cli/src/cmd/forge/script/mod.rs

```diff
@@ -699,6 +713,24 @@ impl ScriptConfig {
     fn target_contract(&self) -> &ArtifactId {
         self.target_contract.as_ref().expect("should exist after building")
     }
+
+    /// Checks if the RPCs used point to chains that support EIP-3855.
+    /// If not, warns the user.
+    async fn check_shanghai_support(&self) -> eyre::Result<()> {
+        for rpc in &self.total_rpcs {
+            let provider = ethers::providers::Provider::<Http>::try_from(rpc)?;
+            let chain_id = provider.get_chainid().await?;
+            if !SHANGHAI_ENABLED_CHAINS.contains(&chain_id) {
+                shell::println("EIP-3855 is not supported in one or more of the RPCs used.")?;
```

Comment:
If this is a warning message, how about a yellow text?
```rust
shell::println(format!("{}", Paint::yellow("EIP-3855 is not supported...")))?;
shell::println(format!("{}", Paint::yellow("Contracts deployed with a ...")))?;
shell::println(format!("{}", Paint::yellow("For more information, pleas...")))?;
```

Reply:
Looks sick
<img width="685" alt="Screen Shot 2023-05-02 at 19 03 37" src="https://user-images.githubusercontent.com/26014927/235803686-da0be525-d08d-437f-a7e7-830f7c0434bf.png">